src/sagemaker/debugger/profiler_config.py

```diff
@@ -81,6 +84,11 @@ class and SageMaker Framework estimators.
         self.framework_profile_params = framework_profile_params
         self.disable_profiler = disable_profiler
 
+        if self.framework_profile_params is not None:
+            logger.warning(
+                "Framework profiling will be deprecated from tensorflow 2.12 and pytorch 2.0"
+            )
```

Comment:
I think build-in mechanism to print deperecation warning exists: https://github.com/aws/sagemaker-python-sdk/blob/5f9d0242b2da712507b0664b41335721369f4417/src/sagemaker/deprecations.py#L56